scene/2d/tile_map.cpp

```diff
@@ -2273,7 +2273,12 @@ void TileMapLayer::force_update() {
 void TileMapLayer::fix_invalid_tiles() {
 	RBSet<Vector2i> coords;
 	for (const KeyValue<Vector2i, TileMapCell> &E : tile_map) {
-		TileSetSource *source = *(tile_map_node->get_tileset())->get_source(E.value.source_id);
+		Ref<TileSet> tileset = tile_map_node->get_tileset();
```

Comment:
Thanks for tackling this issue! 

If the TileSet is null, it will stay null for the whole execution of the function. So I believe a better solution would be to return out of the function if the TileSet is not valid. Something like this:

```
void TileMapLayer::fix_invalid_tiles() {
	Ref<TileSet> tileset = tile_map_node->get_tileset();
	ERR_FAIL_COND_MSG(!tile_set.is_valid(), "Cannot call fix_invalid_tiles() on a TileMap without a valid TileSet");

	RBSet<Vector2i> coords;
	for (const KeyValue<Vector2i, TileMapCell> &E : tile_map) {
		TileSetSource *source = tile_set->get_source(E.value.source_id);
		...
	}
```